kernel/rros/c_trace.c

```diff
@@ -0,0 +1,326 @@
+#define CREATE_TRACE_POINTS
+#include <trace/events/rros.h>
+#include <uapi/linux/uio.h>
+
+#define copy_on_stack(dst,src) \
+char dst[src.iov_len+1]; \
+strncpy(dst,src.iov_base,src.iov_len); \
+dst[src.iov_len] = 0;
```

Comment:
Make this in right format.
```
#define copy_on_stack(dst,src) \
    char dst[src.iov_len+1]; \
    strncpy(dst,src.iov_base,src.iov_len); \
    dst[src.iov_len] = 0;
```

Reply:
By the way, the variable length array (VLA) is discouraged in the kernel (although not prohibited).  I think it will be better to use the C-style string in the rust structure when it might interact with linux-binding functions.

Reply:
Thanks for the reminder. I will write a tutorial, format the C code and test the trace command in libevl.

Reply:
test